scale/scale-codegen/src/main/java/com/strategyobject/substrateclient/scale/codegen/writer/WriterCompositor.java

```diff
@@ -107,6 +110,47 @@ protected CodeBlock whenGenericType(@NonNull DeclaredType type, TypeMirror overr
         return builder.add(")").build();
     }
 
+    @Override
+    protected CodeBlock whenArrayTypeVar(@NonNull ArrayType arrayType, @NonNull TypeVariable componentType, TypeMirror override) {
```

Comment:
It won't work correctly. Given class:
```
class Some<T> {
  T[] a;
}
```
When `T` is revealed for `Some<String>` as `String` it will resolve `StringWriter` for the field `a` instead of the writer of arrays.

Reply:
Nice catch, thanks.